examples/with-typescript/pages/pagewithasync.tsx

```diff
@@ -0,0 +1,28 @@
+import * as React from 'react'
+import 'isomorphic-fetch'
+interface pagesProps {
+  stars: number
+}
+interface pageState {
+  clicked: boolean
+}
+export default class extends React.Component<pagesProps, pageState> {
+  static async getInitialProps() {
+    const res = await fetch('https://api.github.com/repos/zeit/next.js');
+    const data = await res.json();
+    return { stars: data.stargazers_count }
+  }
+  constructor() {
+    super();
+    this.state = { clicked: false }
```

Comment:
only set state in constructor if it relies on props. this should be a simple member. 

```ts
public state: pageState = { clicked: false };
```